src/beast/beast/unit_test/reporter.h

```diff
@@ -25,178 +25,313 @@
+        results();
 
         void
-        add (suite_results const& r)
-        {
-            ++suites;
-            total += r.total;
-            cases += r.cases;
-            failed += r.failed;
-        }
+        add (suite_results const& r);
     };
 
-    boost::optional <std_ostream> m_std_ostream;
-    std::reference_wrapper <abstract_ostream> m_stream;
-    results m_results;
-    suite_results m_suite;
-    case_results m_case;
+    boost::optional <std_ostream> std_ostream_;
+    std::reference_wrapper <abstract_ostream> stream_;
+    results results_;
+    suite_results suite_results_;
+    case_results case_results_;
 
 public:
     reporter (reporter const&) = delete;
     reporter& operator= (reporter const&) = delete;
 
-    explicit reporter (std::ostream& stream = std::cout)
-        : m_std_ostream (std::ref (stream))
-        , m_stream (*m_std_ostream)
-    {
-    }
+    ~reporter();
 
-    ~reporter()
-    {
-        m_stream.get() <<
-            amount (m_results.suites, "suite") << ", " <<
-            amount (m_results.cases, "case") << ", " <<
-            amount (m_results.total, "test") << " total, " <<
-            amount (m_results.failed, "failure")
-            ;
-    }
+    explicit
+    reporter (std::ostream& stream = std::cout);
 
     explicit
-    reporter (abstract_ostream& stream)
-        : m_stream (stream)
-    {
-    }
+    reporter (abstract_ostream& stream);
 
 private:
+    static
+    std::string
+    fmtdur (typename clock_type::duration const& d);
+
     virtual
     void
-    on_suite_begin (suite_info const& info) override
-    {
-        m_suite = suite_results (info.full_name());
-    }
+    on_suite_begin (suite_info const& info) override;
 
     virtual
     void
-    on_suite_end() override
-    {
-        m_results.add (m_suite);
-    }
+    on_suite_end() override;
 
     virtual
     void
-    on_case_begin (std::string const& name) override
-    {
-        m_case = case_results (name);
+    on_case_begin (std::string const& name) override;
 
-        m_stream.get() <<
-            m_suite.name <<
-            (m_case.name.empty() ?
-                "" : (" " + m_case.name))
-        ;
-    }
+    virtual
+    void
+    on_case_end() override;
 
     virtual
     void
-    on_case_end() override
-    {
-        m_suite.add (m_case);
-    }
+    on_pass() override;
 
     virtual
     void
-    on_pass() override
-    {
-        ++m_case.total;
-    }
+    on_fail (std::string const& reason) override;
 
     virtual
     void
-    on_fail (std::string const& reason) override
+    on_log (std::string const& s) override;
+};
+
+//------------------------------------------------------------------------------
+
+template <class _>
+reporter<_>::case_results::case_results (
+        std::string const& name_)
+    : name (name_)
+{
+}
+
+template <class _>
+reporter<_>::suite_results::suite_results (
+        std::string const& name_)
+    : name (name_)
+    , start (clock_type::now())
```

Comment:
why not have `clock_type::now()` as the initializer for the start member variable?